src/Integration.Vsix/InfoBar/InfoBarManager.cs

```diff
@@ -132,6 +172,29 @@ private IInfoBar AttachInfoBarImpl(Guid toolWindowGuid, string message, SonarLin
             return null;
         }
 
+        private IInfoBar AttachInfoBarMainWindowImpl(string message, SonarLintImageMoniker imageMoniker, ButtonStyle buttonStyle = ButtonStyle.Button, params string[] buttonTexts)
```

Comment:
Pretty similar to the other implementation except for "shell.GetProperty((int)__VSSPROPID7.VSSPROPID_MainWindowInfoBarHost, out object value);" and I am also adding the info bar to the host directly. here instead of going through the method TryAddInfoBarToFrame()